.github/workflows/build-deploy-pudl.yml

```diff
@@ -26,12 +26,22 @@ jobs:
         if: ${{ (github.event_name == 'schedule') }}
         run: |
           echo "This action was triggered by a schedule." && echo "GCE_INSTANCE=pudl-deployment-dev" >> $GITHUB_ENV && echo "GITHUB_REF=dev" >> $GITHUB_ENV
+          echo "NIGHTLY_TAG=nightly-$(date +%Y-%m-%d)" >> $GITHUB_ENV
+          echo "NIGHTLY_TAG: $NIGHTLY_TAG"
 
       - name: Checkout Repository
         uses: actions/checkout@v4
         with:
           ref: ${{ env.GITHUB_REF }}
 
+      - name: Tag nightly build
+        if: ${{ (github.event_name == 'schedule') }}
+        run: |
+          git config user.email "pudl@catalyst.coop"
+          git config user.name "PudlBot"
+          git tag -a -m "$NIGHTLY_TAG" $NIGHTLY_TAG $GITHUB_REF
+          git push origin $NIGHTLY_TAG
+
```

Comment:
I *think* this is doing what we want it to now.  It's a tag, it's applied to the branch that we're going to build (not `main` which is checked out here).

Reply:
I think that's close, except that we **only** distribute the most recent nightly build and tagged releases to AWS. I imagine we would also have a copy of the most recent release in a `stable/` directory.  So if you want to refer to `nightly-202x-xx-xx` you would need to look at the `gs://builds.catalyst.coop/nightly-202x-xx-xx` path where the last 30 days of nightly build outputs are stashed, regardless of whether they were successful or not. So if you want to look at nightly build outputs + the code that generated them, you can look at the GCS output, and the tagged commit that matches its name.

Reply:
Ah, right - that makes sense! Are there any barriers to distributing a few recent nightly builds on AWS as well as GCS?

Reply:
We have a limited storage quota on AWS, and I don't think we actually want the public looking at our nightly build outputs, which may or or may not correspond to successful runs.